### ISSUE TITLE
createPersistentToken() is actually asynchronous

### ISSUE THREAD

**Issue by undavide** [AdobeDocs/uxp-photoshop#245]
Issue in /src/pages/uxp/reference-js/Modules/uxp/Persistent File Storage/FileSystemProvider.md

In [`createPersistentToken()`](https://www.stage.adobe.io/photoshop/uxp/uxp/reference-js/Modules/uxp/Persistent%20File%20Storage/FileSystemProvider/#createsessiontokenentry) it's said that it returns a String, very much like `createSessionToken()`. 
However `createPersistentToken()` is asynchronous, and it returns a Promise. In the doc it should be made clearer, with the use of `await`.
Besides, is it as designed that creating a session token is a synch operation and a permanent one is async? 
Thanks!

Davide

**Comment by ErinFinnegan** [comment]
I'm gonna tag @pklaschka for this one...

**Comment by undavide** [comment]
Is he working for the red A now? 😅

**Comment by pklaschka** [comment]
> Is he working for the red A now? 😅

Yep, for the next year, I'll try my best to help "from the other side", as a working student within the Creative Cloud Platform & Ecosystem Team. I'll take a look at this as soon as I can find the time 😉

**Comment by undavide** [comment]
Super cool Pablo, congratulations!

**Comment by pklaschka** [comment]
A few observations (both as notes for myself and to keep you up to date):

1. This particular piece of documentation seems to **also** be managed in [the general UXP reference documentation](https://github.com/AdobeDocs/uxp/blob/main/src/pages/uxp/reference-js/Modules/uxp/Persistent%20File%20Storage/FileSystemProvider.md). I will try to find out where this issue, therefore, belongs
2. I was able to confirm the described behaviour: `createSessionToken` immediately returns a string, while `createPersistentToken` (differing from the documentation) actually returns a promise for that string, therefore, the proper return type should, in fact, be `Promise<string>`.
3. While I don't know if this is, in fact, intended API-design wise, I guess it (technically) makes sense: To create a session token, a lookup would only have to be stored within the current "runtime", so to speak, while having a persistent token presumably requires writing a lookup for that token to the file system (and file i/o needing an async operation, at least, sounds plausible)

I have tested all this using Adobe XD v42.0.22.11, but think we can just assume (and trust @undavide ) that this behaviour is probably the same for various versions of UXP (may that be in PS, XD, or somewhere else).

Since I don't know (yet) how the reference part of the documentation gets updated (I could imagine it getting auto-generated, in which case, a simple PR fixing it won't get us very far), I'll try to find out how (and where 😉 ) we can get this updated. Thank you for reporting this 👍 🙂

**Comment by undavide** [comment]
You're welcome @pklaschka, thanks for looking into it!

**Comment by pklaschka** [comment]
Ok... Here's what I have so far:

1. There seems to be a more general problem with Promises within the documentation, right now (e.g., `getFileForSaving` is also documented to return a file while it returns a promise for it, I'm not sure if it's every function, but at least this is not a "singular" problem).
2. The documentation for the UXP module is maintained (as suspected) within https://github.com/AdobeDocs/uxp (or, more specifically, in this case, https://github.com/AdobeDocs/uxp/blob/main/src/pages/uxp/reference-js/Modules/uxp/Persistent%20File%20Storage/FileSystemProvider.md). They are only "referenced"/"imported" (I don't know what the correct terminology is for Gatsby-based pages) in the XD and Photoshop specific documentations (like this repo). I'll therefore (I hope you don't mind, @undavide ) transfer this issue over to there (since we need to update it there for it to get changed here).
3. This is also related to at least one issue there: https://github.com/AdobeDocs/uxp/issues/53
4. I'm now looking into how we can update this part of the documentation (from what I understand, it's a little more complicated, unfortunately, since it's auto-generated => not maintained within the GitHub repo)

---

Issue was originally https://github.com/AdobeDocs/uxp-photoshop/issues/245; Transferring to https://github.com/AdobeDocs/uxp

Pinging: @kerrishotts
